src/vs/workbench/parts/preferences/browser/settingsEditor2.ts

```diff
@@ -53,11 +54,21 @@ export class SettingsEditor2 extends BaseEditor {
 
 	public static readonly ID: string = 'workbench.editor.settings2';
 	private static NUM_INSTANCES: number = 0;
+	private static SETTING_UPDATE_FAST_DEBOUNCE: number = 1000;
+	private static SETTING_UPDATE_SLOW_DEBOUNCE: number = 200;
 
 	private static readonly SUGGESTIONS: string[] = [
 		'@modified', '@tag:usesOnlineServices'
 	];
 
+	private static shouldSettingUpdateFast(type: SettingValueType | SettingValueType[]): boolean {
+		if (isArray(type)) {
+			// nullable integer/number or complex
+			return true;
+		}
+		return type !== SettingValueType.Enum && type !== SettingValueType.Complex;
```

Comment:
Why should enum be slow?

Reply:
Misnamed the constants (fast should be be named slow and vice versa), will fix

Reply:
My logic was negated here (also the naming of the debounces). *Sigh*, That's what happens when you don't get enough sleep. Will fix